integration/tests/cook/test_basic.py

```diff
@@ -675,7 +675,7 @@ def memory_limit_exceeded_helper(self, command, executor_type, mem=128):
             job_details = f"Job details: {json.dumps(job, sort_keys=True)}"
             self.assertEqual('failed', job['state'], job_details)
             self.assertLessEqual(1, len(job['instances']), job_details)
-            instance = job['instances'][-1]
+            instance = next(i for i in job['instances'] if i['reason_code'] in [2002, 99003])
```

Comment:
Instead of grabbing the "last" instance, grab the instance with the expected instance code.

We should never assume that a particular instance (by index) of a job is the one we're interested in.